web/models/package.ex

```diff
@@ -119,19 +119,41 @@ defmodule HexWeb.Package do
     query
   end
 
-  defp search(query, search) do
-    name_search = escape_search(search)
-    name_search = if String.length(search) >= 3, do: "%" <> name_search <> "%", else: name_search
+  defp search(query, search) when is_binary(search) do
+    filter =
+      if String.length(search) >= 3 do
+        {:contains, search}
+      else
+        {:equals, search}
+      end
```

Comment:
This is just because the final search/2 function expects a {mode, string} pair to specify how to search.I had put this decision in the controller initially, but I had to move it back into the model as there is a controller in the API which also uses the function.